include/deal.II/base/mpi.h

```diff
@@ -1538,6 +1539,85 @@ namespace Utilities
 #  endif
     }
 
+    namespace internal
+    {
+      template <typename T>
+      ArrayView<T>
+      builtin_broadcast(const MPI_Comm &    comm,
+                        const ArrayView<T> &object_to_send,
+                        const unsigned int  root_process)
+      {
+#  ifndef DEAL_II_WITH_MPI
+        (void)comm;
+        (void)root_process;
+        return object_to_send;
+#  else
+        MPI_Datatype datatype =
+          dealii::Utilities::MPI::internal::mpi_type_id(object_to_send);
+
+        ArrayView<T> buffer;
+        if (this_mpi_process(comm) == root_process)
+          {
+            buffer = object_to_send;
+          }
+
+        int ierr =
+          MPI_Bcast(&buffer, buffer.size(), datatype, root_process, comm);
+        AssertThrowMPI(ierr);
+
+        return buffer;
+#  endif
+      }
+
+      template <typename T>
+      T
+      builtin_single_broadcast(const MPI_Comm &   comm,
+                               const T &          object_to_send,
+                               const unsigned int root_process)
+      {
+#  ifndef DEAL_II_WITH_MPI
+        (void)comm;
+        (void)root_process;
+        return object_to_send;
+#  else
+        MPI_Datatype datatype =
+          dealii::Utilities::MPI::internal::mpi_type_id(object_to_send);
+
+        T buffer;
+        if (this_mpi_process(comm) == root_process)
+          {
+            buffer = object_to_send;
+          }
+
+        int ierr = MPI_Bcast(&buffer, 1, datatype, root_process, comm);
+        AssertThrowMPI(ierr);
+
+        return buffer;
+#  endif
+      }
+
+    } // namespace internal
+
+    template <typename T>
+    std::enable_if_t<is_mpi_type<T>::value, T>
+    broadcast(const MPI_Comm &   comm,
+              const T &          object_to_send,
+              const unsigned int root_process)
```

Comment:
For data types that *are* MPI types, this function is identical to the one that now starts in line 1623 below of your file. You will get compiler errors for it. To avoid this, you will have to add the opposite `std::enable_if_t<!is_mpi_type<T>::value, T>` (not the `!`) to the other function declaration.

Reply:
Fixed in 147b71fd64585107f0780dbb0826814473a6d371